go/logger/buffer.go

```diff
@@ -88,13 +88,10 @@ func (writer *autoFlushingBufferedWriter) backgroundFlush() {
 
 			writer.backupWriter.Flush()
 		case <-writer.shutdown:
-			writer.timer.shutdownCh <- struct{}{}
+			close(writer.timer.shutdownCh)
```

Comment:
Why are these changes here?

Reply:
required to make shutdown re-entrant

Reply:
I think you likely need this because of the other change which I think you should remove

Reply:
ok cool